h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -1499,4 +1519,49 @@ private void giveDatasetFeedback(Frame frame, UserFeedback userFeedback, HashMap
   private String getModelType(Model m) {
     return m._key.toString().startsWith("XRT_") ? "XRT" : m._parms.algoName();
   }
+  
+  private void cleanUpStackedEnsemblesCVPreds(Job<StackedEnsembleModel> ensembleJob) {
+    Log.info("Remove CV Preds & Models from " + ensembleJob.get()._key.toString() + "'s metalearner");
+    //Clear out all CV preds and CV models
+    if (ensembleJob.get()._output._metalearner._output._cross_validation_predictions != null) {
+      for (Key k : ensembleJob.get()._output._metalearner._output._cross_validation_predictions) {
+        Log.info("Remove CV Predictions for " + ensembleJob.get()._output._metalearner._key.toString());
+        k.remove();
+      }
+    }
+    if (ensembleJob.get()._output._metalearner._output._cross_validation_holdout_predictions_frame_id != null) {
+      Log.info("Remove CV Prediction Frame for " + ensembleJob.get()._output._metalearner._key.toString());
+      ensembleJob.get()._output._metalearner._output._cross_validation_holdout_predictions_frame_id.remove();
+    }
+  }
+
+  private void cleanUpStackedEnsemblesCVModels(Job<StackedEnsembleModel> ensembleJob) {
+    Log.info("Remove CV Models for " + ensembleJob.get()._output._metalearner._key.toString());
+    ensembleJob.get()._output._metalearner.deleteCrossValidationModels();
+  }
+
+  private void cleanUpModelsCVPreds() {
+    //Clear out all CV preds and CV models
+    for (Model model : leaderboard().getModels()) {
+      if (!model._parms.algoName().equals("stackedensemble")) {
+        if (model._output._cross_validation_predictions != null) {
```

Comment:
In my opinion, SE should implement this particular functionality. There should not be any "IFs"

Reply:
Yes, agree. I will need to investigate this after if that is okay.

Reply:
Will address in this PR